--- ifs.build.orig.orig	2026-08-26 14:45:30.485700012 -0400
+++ ifs.build.new	2026-08-26 14:51:49.205685868 -0400
@@ -14,7 +14,7 @@

 [image=20m]
 [virtual=x86_64,multiboot ] boot = {
-    startup-x86 -D8250..115200 -zz  
+    startup-x86 -D8250..115200 -zz -f 0,0,1000M
     # Setting PATH and LD_LIBRARY_PATH here allows them to be set both for
     # procnto as well as programs run from the IFS startup script.
     PATH=/proc/boot:/system/bin
