--- ifs.build.orig	2026-07-05 18:39:54.453013998 -0400
+++ ifs.build	2026-07-05 19:07:00.283338779 -0400
@@ -23,8 +23,12 @@
 }
 
 [+script] startup-script = {
+    procmgr_symlink /proc/boot/ksh /bin/ksh
+    procmgr_symlink /proc/boot/ksh /bin/sh
+    procmgr_symlink /data/var/tmp /tmp
     procmgr_symlink /proc/boot/ldqnx-64.so.2 /usr/lib/ldqnx-64.so.2
 
+    procmgr_symlink /system/etc/terminfo /usr/share/terminfo
     TERM=qansi
     HOME=/data/home/root
     PCI_BKWD_COMPAT_MODULE=/proc/boot/pci_bkwd_compat.so
@@ -258,16 +262,15 @@
 [uid=0 gid=0 perms=0444] qcrypto.conf={
 openssl-3 tags=*
 }
-
-[type=link] /bin/ksh=/proc/boot/ksh
-[type=link] /bin/sh=/proc/boot/ksh
-[type=link] /bin/cp=/proc/boot/toybox
-[type=link] /bin/rm=/proc/boot/toybox
-[type=link] /usr/bin/ssh=/system/bin/ssh
-[type=link] /tmp=/data/var/tmp
-[type=link] /usr/share/terminfo=/system/etc/terminfo
-[type=link] /etc=/data/var/etc
-[type=link] /var=/data/var
+#[type=link] /bin/ksh=/proc/boot/ksh
+#[type=link] /bin/sh=/proc/boot/ksh
+#[type=link] /bin/cp=/proc/boot/toybox
+#[type=link] /bin/rm=/proc/boot/toybox
+#[type=link] /usr/bin/ssh=/system/bin/ssh
+#[type=link] /tmp=/data/var/tmp
+#[type=link] /usr/share/terminfo=/system/etc/terminfo
+#[type=link] /etc=/data/var/etc
+#[type=link] /var=/data/var
 gpio-rp1=bin/gpio-rp1
 msix-rp1=bin/msix-rp1
 wdtkick=bin/wdtkick
